src/transactions/cpfp.rs

```diff
@@ -0,0 +1,181 @@
+use crate::{
+    error::*,
+    transactions::{utils, CpfpableTransaction, RevaultTransaction, CPFP_MIN_CHANGE},
+    txins::*,
+    txouts::*,
+};
+
+use miniscript::bitcoin::{
+    consensus::encode::Decodable,
+    util::psbt::{Input as PsbtIn, PartiallySignedTransaction as Psbt},
+    Amount, Script, SigHashType, Transaction, TxIn, TxOut,
+};
+
+#[cfg(feature = "use-serde")]
+use {
+    serde::de::{self, Deserialize, Deserializer},
+    serde::ser::{Serialize, Serializer},
+};
+
+impl_revault_transaction!(
+    CpfpTransaction,
+    doc = "The transaction feebumping either an unvault or a spend"
+);
+impl CpfpTransaction {
+    /// Create a new cpfp transaction from the transaction that needs cpfp, its fees and weight;
+    /// the required feerate (expressed in sats/WU); a list of CpfpTxIn that can be used along
+    /// with their child number, the cpfp descriptor and the secp curve.
+    pub(crate) fn from_tx(
+        to_be_cpfped: &impl CpfpableTransaction,
+        feerate: f64,
+        mut listunspent: Vec<CpfpTxIn>,
+    ) -> Result<CpfpTransaction, TransactionCreationError> {
+        assert!(to_be_cpfped.is_finalized());
+        // Checking that the feerate requested is higher than the previous
+        // tx feerate (otherwise why cpfping in the first place?)
+        let to_be_cpfped_weight = to_be_cpfped.clone().into_tx().get_weight();
```

Comment:
Ouch, no need to clone there

Reply:
Actually I do, as `into_tx` destroys self, and I can't call `tx` as it doesn't finalize the transaction (it gives you the unsigned one).

I can ask for a `impl CpfpableTransaction` (instead of `&impl CpfpableTransaction` to avoid "hiding" the clone. What do you think?

Reply:
> I can ask for a impl CpfpableTransaction (instead of &impl CpfpableTransaction) to avoid "hiding" the clone. What do you think?

I actually went with this as it's more explicit.

Reply:
I agree for not hiding clones, much better API. I overlooked the requirement for the tx to be finalized that"s.. Surprising. At least it's different from how we've been doing it for all the rest.

Reply:
Actually i don't think it makes sense to require it to be finalized. Just take the unsigned tx weight + the satisfaction weight.